homeassistant/components/stream/hls.py

```diff
@@ -78,21 +78,28 @@ def render_preamble(track):
     @staticmethod
     def render_playlist(track):
         """Render playlist."""
-        segments = track.segments[-NUM_PLAYLIST_SEGMENTS:]
+        segments = list(track.get_segment())[-NUM_PLAYLIST_SEGMENTS:]
 
         if not segments:
             return []
 
-        playlist = ["#EXT-X-MEDIA-SEQUENCE:{}".format(segments[0])]
+        playlist = ["#EXT-X-MEDIA-SEQUENCE:{}".format(segments[0].sequence)]
+        if segments[0].stream_id:
```

Comment:
Don't all the segments have stream_id? And the first stream_id starts at 1 since we are calling `segment_buffer.discontinuity()` every time the worker starts

Reply:
Oh sorry, just saw it only gets called after the worker exits so it starts at 0

Reply:
But do we need to check stream_id here? Can we just include the tag regardless?

Reply:
It is initialized to 0 then `discontinuity()` is only invoked after the `stream_worker` exits.  This means that this class depends on the implementation of `SegmentBuffer` to work.   (I also thought this was the most subtle line of code in this PR, so I think we should make this more clear)

Some thoughts about how to improve readability are:
- add a comment here stating what I just said
- add some state in `StreamOutput` to track the first `stream_id` seen, then only display this on the second one seen

Open to other suggestions to improve readability.

Reply:
When i read [6.2.2 of the rfc](https://tools.ietf.org/html/rfc8216#section-6.2.2) i got the impression we want to add `EXT-X-DISCONTINUITY-SEQUENCE` whenever an instance of `#EXT-X-DISCONTINUITY` is dropped off the playlist.

However, i see that it says you MUST do that, but it doesn't say that it can only be present when that happens.  I'll do some local testing and see what happens with this always included.

It might not hurt.

Reply:
[4.3.3.3](https://tools.ietf.org/html/rfc8216#section-4.3.3.3) says it's considered 0 if it's not included, so it looks like it shouldn't hurt

Reply:
Great, I've made this change and tested it out and it worked great.  Done.